src/Engine/ProtoCore/Reflection/MemberInfoReflectionExtensions.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Autodesk.DesignScript.Runtime;
+
+namespace ProtoCore.Reflection
+{
+    internal static class MemberInfoReflectionExtensions
+    {
+        internal static MethodInfo GetBaseDefinitionReflectionContext(this MethodInfo methodInfo)
+        {
+            try
+            {
+                // if overriding object 
+                var baseType = methodInfo.DeclaringType.BaseType;
+                if (baseType is null)
+                {
+                    return methodInfo;
+                }
+
+                // This will fail if there are overloads of the method
+                var baseTypeMethod = baseType.GetMethod(methodInfo.Name);
+
+                if (baseTypeMethod is null)
+                {
+                    return methodInfo;
+                }
+
+                // Need to return the original methodInfo if the method is hiding the inherited member using the new keyword
+                // This is probably not a supported way of doing this... but getting desperate
+                if (baseTypeMethod.IsVirtual != true && baseTypeMethod.GetMethodBody() != null)
```

Comment:
what libraries did you see this in?

Reply:
I came across it a few times in the Revit library https://github.com/DynamoDS/DynamoRevit/blob/master/src/Libraries/RevitNodes/Elements/FamilyInstance.cs#L318

Essentially this extension needs to replicate this https://docs.microsoft.com/en-us/dotnet/api/system.reflection.methodinfo.getbasedefinition?view=netframework-4.8#remarks for reflection only types